src/compiler/crystal/types.cr

```diff
@@ -655,7 +655,7 @@ module Crystal
     # Adds a location to this type.
     def add_location(location : Location)
       locations = @locations ||= [] of Location
-      locations << location
+      locations << location unless locations.any? { |loc| loc.filename == location.filename && loc.line_number == location.line_number }
```

Comment:
What's the reason for adding this condition? Is there a likely case for a type to be defined multiple times in the same line?

Reply:
This should prevent that generations like in https://crystal-lang.org/api/0.25.0/Float32.html happen where 2 exactly same links are generated.

Reply:
Ah, it's because of macro expansion. Yeah, then it's probably better to leave this in.
I'd add a comment about this, though.